pkg/reconciler/kubernetes/tektonhub/tektonhub.go

```diff
@@ -0,0 +1,595 @@
+	// Create new, empty manifest with valid client and logger
+	installed := r.manifest.Append()
+	stages := common.Stages{common.AppendInstalled}
+	err := stages.Execute(ctx, &installed, instance)
+	return &installed, err
+}
+
+// checkIfInstallerSetExist checks if installer set exists for a component and return true/false based on it
+// and if installer set which already exist is of older version then it deletes and return false to create a new
+// installer set
+func checkIfInstallerSetExist(ctx context.Context, oc clientset.Interface, relVersion string,
+	th *v1alpha1.TektonHub, component string) (bool, error) {
+
+	// Check if installer set is already created
+	compInstallerSet, ok := th.Status.HubInstallerSet[component]
+	if !ok {
+		return false, nil
+	}
+
+	if compInstallerSet != "" {
+		// if already created then check which version it is
+		ctIs, err := oc.OperatorV1alpha1().TektonInstallerSets().
+			Get(ctx, compInstallerSet, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return false, nil
+			}
+			return false, err
+		}
+
+		version, ok := ctIs.Annotations[releaseVersionKey]
+		if ok && version == relVersion {
+			// if installer set already exist and release version is same
+			// then ignore and move on
+			return true, nil
+		}
+
+		// release version doesn't exist or is different from expected
+		// deleted existing InstallerSet and create a new one
+
+		err = oc.OperatorV1alpha1().TektonInstallerSets().
+			Delete(ctx, compInstallerSet, metav1.DeleteOptions{})
+		if err != nil {
+			return false, err
+		}
+	}
+
+	return false, nil
+}
+
+func createInstallerSet(ctx context.Context, oc clientset.Interface, th *v1alpha1.TektonHub,
+	manifest mf.Manifest, releaseVersion, component, installerSetPrefix, namespace string) error {
+
+	is := makeInstallerSet(th, manifest, installerSetPrefix, releaseVersion, namespace)
+
+	createdIs, err := oc.OperatorV1alpha1().TektonInstallerSets().
+		Create(ctx, is, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	if len(th.Status.HubInstallerSet) == 0 {
+		th.Status.HubInstallerSet = map[string]string{}
+	}
+
+	// Update the status of addon with created installerSet name
+	th.Status.HubInstallerSet[component] = createdIs.Name
+	th.Status.SetVersion(releaseVersion)
+
+	_, err = oc.OperatorV1alpha1().TektonHubs().
+		UpdateStatus(ctx, th, metav1.UpdateOptions{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func makeInstallerSet(th *v1alpha1.TektonHub, manifest mf.Manifest, prefix, releaseVersion, namespace string) *v1alpha1.TektonInstallerSet {
+	ownerRef := *metav1.NewControllerRef(th, th.GetGroupVersionKind())
+	return &v1alpha1.TektonInstallerSet{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", prefix),
+			Labels: map[string]string{
+				createdByKey: createdByValue,
+			},
+			Annotations: map[string]string{
+				releaseVersionKey:  releaseVersion,
```

Comment:
we need additional labels here:
- type
- version (operator version)
- createdBy

eg: https://github.com/tektoncd/operator/blob/a0cd67e5c428babf1f76316a14cf9c4526055cab/pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go#L359